helix-core/src/main/java/org/apache/helix/controller/stages/PerReplicaThrottleStage.java

```diff
@@ -146,8 +153,8 @@ private MessageOutput compute(ClusterEvent event, Map<String, Resource> resource
           partitonMsgMap.put(partition, msgList);
         }
         MessageOutput resourceMsgOut =
-            throttlePerReplicaMessages(idealState, partitonMsgMap, bestPossibleStateOutput,
-                throttleController, retracedPartitionsState);
+            throttlePerReplicaMessages(idealState, currentStateOutput, partitonMsgMap,
+                bestPossibleStateOutput, dataCache, throttleController, retracedPartitionsState);
```

Comment:
let me see how long the parameter list would be.

cache.getStateModelDef(stateModelDefName);
cache.getEnabledLiveInstances()

This one is deadly, as we need to get per partition. We have to have cache in throttlePerReplicaMessages.

cache
        .getDisabledInstancesForPartition(idealState.getResourceName(),
            partition.getPartitionName())

Let reduce passing cache to some private method called by `throttlePerReplicaMessages` if the don't call getDisabledInstancesForPartition . But I can't get rid of cache here. Some private method do need it.

Reply:
We can't really get rid of it here. But I did cleaned all the path inside throttlePerReplicaMessages so that it won't pass cache around unless we need call the above api